File: UDPClient/Assets/Packet/SCPacketOtherMove.cs
using UnityEngine;
using static BinaryUtility;

public class SCPacketOtherMove : UDPPacket
{
	public int mPlayerID;
	public float mPosX;
	public float mPosY;
	public override void read(byte[] buffer, ref int index)
	{
		base.read(buffer, ref index);
		mPlayerID = readInt(buffer, ref index, out _);
		mPosX = readFloat(buffer, ref index, out _);
		mPosY = readFloat(buffer, ref index, out _);
	}
	public override void execute()
	{
		GameFramework.get().setPlayerPosition(mPlayerID, new Vector3(mPosX, mPosY));
	}
}

File: UDPServer/Packet/SCPacketOtherMove.cpp
#include "SCPacketOtherMove.h"

void SCPacketOtherMove::write(char* buffer, int bufferSize, int& index)
{
	base::write(buffer, bufferSize, index);
	Utility::write(buffer, bufferSize, index, mPlayerID);
	Utility::write(buffer, bufferSize, index, mPosX);
	Utility::write(buffer, bufferSize, index, mPosY);
}